### DISCUSSION TITLE
Creating a $HOME/.netrc with an action

### DISCUSSION THREAD

**Discussion by dinoshauer** [community/community#25836] [Actions]
<p>Hello! I’ve created an action that creates a .netrc file for curl to use (<a href="https://github.com/onomondo/github-auth-netrc-javascript-action" rel="nofollow noopener">https://github.com/onomondo/github-auth-netrc-javascript-action</a>) to authenticate with github when we run npm install as we have a package defined as a private repo.</p>
<p>The thing is it works if I do it manually in a step:</p>
<pre><code>name: CI

on: [push]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: npm install, build and test
      run: |
        echo "machine github.com login ${{ secrets.GITHUB_AUTH_TOKEN }} password x-oauth-basic" &gt; $HOME/.netrc
        npm install
        npx vue-cli-service lint
</code></pre>
<p>Here’s a configuration using the action that does not work:</p>
<pre><code>on: [push]

jobs:
  hello_world_job:
    runs-on: ubuntu-latest
    name: Install private npm package hosted on github
    steps:
    - uses: actions/checkout@v1
    - uses: onomondo/github-auth-netrc-javascript-action@master
      with:
        github-auth-token: ${{ secrets.GITHUB_AUTH_KEY }}
    - name: npm install, build and test
      run: |
        npm install
</code></pre>
<p>I’ve tried listing the home directory etc both in the action and in the run steps and the file is listed in the output.</p>
<p>Any clues whether this is a limitation of actions and I’ll have to continue doing it inline or?</p>

**Comment by chrispat** [comment, marked as answer]
<p><a class="mention" href="/dinoshauer">@dinoshauer</a> In the example you have below you have two different secret names.  Is that intentional?</p>

**Comment by dinoshauer** [comment]
<p>Wow. Thanks for the fresh set of eyes!</p>
